lib/src/base/states/filterable_state_base.dart

```diff
@@ -1,7 +1,15 @@
 part of 'index.dart';
 
+/// Is an interface class which serves as a base for the [NetworkFilterableState].
+///
+/// Inherits [data] and [status] from the [NetworkStateBase], and provides the new [visibleData] and [filter] properties.
+///
 abstract class NetworkFilterableStateBase<T, F> extends NetworkStateBase<T> {
+  /// Holds the data of type `T`, that will be displayed in the UI based on the applied [filter].
```

Comment:
Maybe write : `be used to locally filter elements and separate all data from filtered data`